### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by raldone01**
This commit adds a simple workflow to publish docker images to `ghcr.io`.

Currently it runs the python script with root.
It would be better to run as some other user like `33:33`.
A simple `docker-compose.yml` is also provided.

In docker it is customary to run services in different containers.
I run ollama in a container under the url `http://ollama`.
I found no way to specify fully custom base urls without modifying the python script or using the sdk.

It builds all images and seems to work.
I don't have the best insight in the code base so please test it.

### CODE REVIEW

**Comment by iojw** [comment]
Thanks for this PR! Could you share more about your use case and when this is needed?

**Comment by raldone01** [comment]
I am not sure what exactly you are asking for.

# Regarding docker

I am sure there are some good blogs about the advantage of docker out on the internet.
I will break down the key points:

* No modification of the environment of the host machine. Users don't have to deal with the python environment and installing the software. They just have to copy and adapt the compose file. They don't have to know anything about the software installation process.

* Predictable environment. All users have the same dependencies and python version installed.

* Better isolation. Since docker containers can't access arbitrary files/processes on the host.

# Regarding the custom base url:

Basically I have ollama running on a different machine under some domain like `ollama.custom.com` and would like to pass the base uri to the router.

**Comment by iojw** [comment]
Thank you for elaborating! I definitely understand the advantages of using Docker and I think it's great that you're using it. However, I don't think it makes sense to publish a universal Docker image for RouteLLM right now since the specific requirements (drivers, deps, version, base image etc.) will differ based on use-case. 

The fact that routellm can be installed from PyPI makes it very easy for anyone to build a Docker image that suits their own needs. I hope this makes sense, let me know if you have any other thoughts.

**Comment by raldone01** [comment]
Makes sense. I will just create a routllm-docker repo and build and host it from there.
